src/Psalm/Internal/Analyzer/FileAnalyzer.php

```diff
@@ -150,9 +150,9 @@ public function analyze(
             return;
         }
 
-        $event = new BeforeFileAnalysisEvent($this, $this->context, $file_storage, $codebase);
-
+        $event = new BeforeFileAnalysisEvent($this, $this->context, $file_storage, $codebase, $stmts);
         $codebase->config->eventDispatcher->dispatchBeforeFileAnalysis($event);
+        $stmts = $event->getStmts();
```

Comment:
This implies the `$stmts` array may be changed by the handler, but there's no method on BeforeAnalysisEvent to set the updated array of statements.

Note that plugins may still change the statement objects in this array - there's no need to fetch statements from the event to support that.

Reply:
I'd be fine adding a `setStmts(array $stmts)` method to the event as well - or remove the line above from `FileAnalyzer` (since the references of inside that array are kept).

My feeling is, that dropping the line from above and not introducing `setStmts(array $stmts)` has less side effects.